arch/arm/core/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -52,4 +52,37 @@ int _arch_mem_domain_max_partitions_get(void)
 {
 	return arm_core_mpu_get_max_domain_partition_regions();
 }
+
+/**
+ * @brief Reset MPU region for a single memory partition
+ *
+ * @param   domain  The memory domain structure
+ * @param   part_index  memory partition index
+ */
+void _arch_mem_domain_remove_partition(struct k_mem_domain *domain,
+				       u32_t  partition_id)
+{
+	ARG_UNUSED(domain);
+
+	arm_core_mpu_disable();
```

Comment:
Why do we need these functions to do anything?
My understanding was that the additional _arch_mem_domain* APIs were needed for the x86 MMU, that the ARM implementation is already working fine.
So why aren't these no-ops for ARM?

Reply:
If we create an application on ARM and move it to x86 there will be inconsistencies between the working of the kernel itself. In ARM everything will be  handled in the swap and this is what the user might expect when he ports it to x86. But that will not be the case and he might see his application break at some point.